swap_meet/vendor.py

```diff
@@ -1,2 +1,76 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        if inventory is None:
+            self.inventory = [] 
+        else:
+            self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False
+    
+    def get_by_id(self, item_id: int):
+        for item in self.inventory:
+            if item.id == item_id:
+                return item
+        return None
+    
+    def swap_items(self, other_vendor, my_item, their_item):
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
```

Comment:
With longer compound conditionals such as this, keep in mind our PEP 8 requirements that lines be under 79 characters long! If you start to suspect you are getting close to that limit, feel free to break up the conditional statement across multiple lines. If you don't have parentheses or brackets to help Python intuit that multiple lines belong to the same statement, we can use the backslash (\) to indicate line continuation.

```suggestion
        if my_item not in self.inventory \
            or their_item not in other_vendor.inventory:
```